src/ppx/application.cpp

```diff
@@ -810,13 +810,13 @@ void Application::InitStandardKnobs()
         "Enable metrics report output. See also: `--metrics-filename` and `--overwrite-metrics-file`.");
 
     mStandardOpts.pFrameCount =
-        mKnobManager.CreateKnob<KnobFlag<uint64_t>>("frame-count", 0, 0, mSettings.standardKnobsDefaultValue.frameCount);
+        mKnobManager.CreateKnob<KnobFlag<uint64_t>>("frame-count", mSettings.standardKnobsDefaultValue.frameCount, 0, UINT64_MAX);
```

Comment:
One way to do that, but it's a bit verbose is to have a specific type to wrap the integers/types:

```cpp
template<typename T>
struct DefaultValue {
  T value_;

private:
  DefaultValue(T value) : value_(value) { }

public:
static
  T Make(T value) {
    return DefaultValue(value);
  }

  operator T() const { return value_; }
};

KnobSlider(const std::string& flagName, DefaultValue<T> defaultValue, T minValue, T maxValue);
```

And usage is:

```cpp
k3  = km.CreateKnob<ppx::KnobSlider<int>>("flag_name3", DefaultValue<int>::Make(5), 0, 10);
```

But that's a lot of text just to pass around 5...

Reply:
I think I prefer the idea of having default min/max as `INT_MIN` `INT_MAX` etc. and then allowing people to overwrite it with `SetMin()` `SetMax()`. I think this could also work with the adjustable min/max stuff I wanted to work on.